watertap/unit_models/tests/test_cstr_injection.py

```diff
@@ -338,184 +337,71 @@ def test_get_performance_contents(self, sapon):
-    @pytest.mark.skipif(solver is None, reason="Solver not available")
-    @pytest.mark.component
-    def test_conservation(self, model):
-        assert (
-            abs(
-                value(
-                    model.fs.unit.inlet.flow_vol[0] - model.fs.unit.outlet.flow_vol[0]
-                )
-            )
-            <= 1e-6
-        )
-        assert abs(
-            value(
-                model.fs.unit.inlet.flow_vol[0]
-                * sum(
-                    model.fs.unit.inlet.conc_mass_comp[0, j]
-                    for j in model.fs.properties.solute_set
-                )
-                - model.fs.unit.outlet.flow_vol[0]
+        self.unit_solutions[
+            (
+                m.fs.unit.outlet.flow_vol[0]
                 * sum(
-                    model.fs.unit.outlet.conc_mass_comp[0, j]
-                    for j in model.fs.properties.solute_set
+                    m.fs.unit.outlet.conc_mass_comp[0, j]
+                    for j in m.fs.properties.solute_set
                 )
                 + sum(
-                    model.fs.unit.control_volume.rate_reaction_generation[0, "Liq", j]
-                    for j in model.fs.properties.solute_set
+                    m.fs.unit.control_volume.rate_reaction_generation[0, "Liq", j]
+                    for j in m.fs.properties.solute_set
                 )
-                <= 1e-6
+            )
+        ] = value(
+            m.fs.unit.inlet.flow_vol[0]
+            * sum(
+                m.fs.unit.inlet.conc_mass_comp[0, j] for j in m.fs.properties.solute_set
             )
         )
 
-    @pytest.mark.solver
-    @pytest.mark.skipif(solver is None, reason="Solver not available")
-    @pytest.mark.component
-    def test_costing(self, model):
-        m = model
+        return m
+
+
+class TestCosting(UnitTestHarness):
+    def configure(self):
+        m = build_ASM1()
 
+        # Add unit model costing
         m.fs.costing = WaterTAPCosting()
 
         m.fs.unit.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
         m.fs.costing.cost_process()
         m.fs.costing.add_LCOW(m.fs.unit.control_volume.properties_out[0].flow_vol)
-        solver = get_solver()
-        results = solver.solve(m)
+        m.objective = Objective(expr=m.fs.costing.LCOW)
 
-        assert_optimal_termination(results)
+        iscale.set_scaling_factor(m.fs.unit.costing.capital_cost, 1e-7)
+
+        iscale.calculate_scaling_factors(m.fs.unit)
+
+        self.unit_solutions[m.fs.unit.costing.capital_cost] = 613502.91662
+
+        m.fs.unit.initialize()
+
+        results = solver.solve(m)
 
-        # Check solutions
-        assert pytest.approx(613502.91662, rel=1e-5) == value(
-            m.fs.unit.costing.capital_cost
-        )
         assert pytest.approx(0.0132455, rel=1e-5) == value(m.fs.costing.LCOW)
 
-    @pytest.mark.unit
-    def test_report(self, model):
-        model.fs.unit.report()
+        return m
 
 
 @pytest.mark.build
```

Comment:
Seems like the ASM2d and modified ASM2d tests below are just making sure the build works. Do we also want to test a few of the solutions for these builds?